src/libData/AccountData/Account.h

```diff
@@ -88,24 +94,27 @@ class Account : public Serializable
     /// Returns the account nonce.
     const uint256_t& GetNonce() const { return m_nonce; }
 
+    void SetStorageRoot(const h256& root);
+
     /// Returns the storage root.
     const h256& GetStorageRoot() const { return m_storageRoot; }
 
-    /// Returns the code hash.
-    const h256& GetCodeHash() const { return m_codeHash; }
+    /// Set the code
+    void SetCode(const std::vector<unsigned char>& code);
 
-    /// Set the code of the account. Used by "create" messages
-    void SetCode(vector<unsigned char>&& code);
+    const std::vector<unsigned char>& GetCode() const { return m_codeCache; }
 
-    const vector<unsigned char>& GetCode() const { return m_codeCache; }
+    /// Returns the code hash.
+    const h256& GetCodeHash() const { return m_codeHash; }
 
-    void SetStorage(string _k, string _mutable, string _type, string _v);
+    void SetStorage(string _k, string _type, string _v, bool _mutable = true);
 
-    vector<string> GetKeys();
+    /// Return the data for a parameter, type + value
+    vector<string> GetStorage(string _k) const;
 
-    vector<string> GetStorage(string _k);
+    Json::Value GetInitJson() const { return m_initValJson; }
```

Comment:
A question here about the return type - is this efficient to return a copy of `m_initValJson`? If possible, we could return a reference here (since you already declared it as const)